extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/JsonMeterRegistryProvider.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.micrometer.runtime.export;
+
+import java.time.Duration;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.micrometer.core.instrument.Clock;
+import io.quarkus.micrometer.runtime.registry.json.JsonMeterRegistry;
+
+@Singleton
+public class JsonMeterRegistryProvider {
+
+    private Integer bufferLength;
+    private Duration expiry;
+
+    @Produces
+    @Singleton
+    public JsonMeterRegistry registry(Clock clock) {
+        return new JsonMeterRegistry(clock, bufferLength, expiry);
+    }
+
+    public void configureJsonRegistry(Integer bufferLength, Duration expiry) {
+        this.bufferLength = bufferLength;
+        this.expiry = expiry;
+    }
```

Comment:
Not sure the configureJsonRegistry or object vars are necessary?

```
    @Produces
    @Singleton
    public JsonMeterRegistry registry(Clock clock, Config config) {
        return new JsonMeterRegistry(clock, config.export.json.bufferLength, config.export.json.expiry);
    }
```

That should just work (injection of microprofile config)?